ethereum/graphqlrpc/src/main/java/tech/pegasys/pantheon/ethereum/graphqlrpc/internal/pojoadapter/BlockAdapterBase.java

```diff
@@ -0,0 +1,238 @@
+    final Hash parentHash = header.getParentHash();
+    final Optional<BlockWithMetadata<TransactionWithMetadata, Hash>> block =
+        query.blockByHash(parentHash);
+    return block.map(NormalBlockAdapter::new);
+  }
+
+  public Optional<Bytes32> getHash() {
+    return Optional.of(header.getHash());
+  }
+
+  public Optional<BytesValue> getNonce() {
+    final long nonce = header.getNonce();
+    final byte[] bytes = Longs.toByteArray(nonce);
+    return Optional.of(BytesValue.wrap(bytes));
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<Bytes32> getTransactionsRoot() {
+    return Optional.of(header.getTransactionsRoot());
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<Bytes32> getStateRoot() {
+    return Optional.of(header.getStateRoot());
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<Bytes32> getReceiptsRoot() {
+    return Optional.of(header.getReceiptsRoot());
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<AccountAdapter> getMiner(final DataFetchingEnvironment environment) {
+
+    final BlockchainQuery query = getBlockchainQuery(environment);
+    long blockNumber = header.getNumber();
+    final Long bn = environment.getArgument("block");
+    if (bn != null) {
+      blockNumber = bn;
+    }
+    return Optional.of(
+        new AccountAdapter(query.getWorldState(blockNumber).get().get(header.getCoinbase())));
+  }
+
+  public Optional<BytesValue> getExtraData() {
+    return Optional.of(header.getExtraData());
+  }
+
+  public Optional<Long> getGasLimit() {
+    return Optional.of(header.getGasLimit());
+  }
+
+  public Optional<Long> getGasUsed() {
+    return Optional.of(header.getGasUsed());
+  }
+
+  public Optional<UInt256> getTimestamp() {
+    return Optional.of(UInt256.of(header.getTimestamp()));
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<BytesValue> getLogsBloom() {
+    return Optional.of(header.getLogsBloom().getBytes());
+  }
+
+  public Optional<Bytes32> getMixHash() {
+    return Optional.of(header.getMixHash());
+  }
+
+  public Optional<UInt256> getDifficulty() {
+    return Optional.of(header.getDifficulty());
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<Bytes32> getOmmerHash() {
+    return Optional.of(header.getOmmersHash());
+  }
+
+  public Optional<Long> getNumber() {
+    final long bn = header.getNumber();
+    return Optional.of(bn);
+  }
+
+  @SuppressWarnings("unused")
+  public Optional<AccountAdapter> getAccount(final DataFetchingEnvironment environment) {
+
+    final BlockchainQuery query = getBlockchainQuery(environment);
+    final long bn = header.getNumber();
+    final MutableWorldState ws = query.getWorldState(bn).get();
+
+    if (ws != null) {
+      final Address addr = environment.getArgument("address");
+      return Optional.of(new AccountAdapter(ws.get(addr)));
+    }
+    return Optional.empty();
+  }
+
+  @SuppressWarnings("unused")
+  public List<LogAdapter> getLogs(final DataFetchingEnvironment environment) {
```

Comment:
The method itself gets marked as unused by IntelliJ, but it is reflected out by GraphQL.  Same for all suppressions of unused in this class.

My concern with creating our own annotation is that GraphQL does not consider annotations internally and an annotation would imply some processing that doesn't happen.  A comment instead of an annotation that this is suppressed because GraphQL reflects it would serve us better.